Source/ASDisplayNode+Layout.mm

```diff
@@ -387,11 +394,14 @@ - (void)_u_measureNodeWithBoundsIfNecessary:(CGRect)bounds
     // In this case, we need to detect that we've already asked to be resized to match this
     // particular ASLayout object, and shouldn't loop asking again unless we have a different ASLayout.
     nextLayout->requestedLayoutFromAbove = YES;
-    __instanceLock__.unlock();
-    [self _u_setNeedsLayoutFromAbove];
-    __instanceLock__.lock();
+    
+    {
+      ASDN::MutexUnlocker u(__instanceLock__);
+      [self _u_setNeedsLayoutFromAbove];
```

Comment:
Apply the same fix in #1022 here just to be safe, i.e if `_u_setNeedsLayoutFromAbove` causes an assertion, it'll be correctly shown.